lib/widgets/grid.dart

```diff
@@ -0,0 +1,195 @@
+import 'dart:io';
+
+import 'package:files/backend/entity_info.dart';
+import 'package:files/backend/utils.dart';
+import 'package:files/widgets/entity_context_menu.dart';
+import 'package:files/widgets/workspace.dart';
+import 'package:flutter/material.dart';
+
+typedef EntityCallback = void Function(EntityInfo entity);
+typedef DropAcceptCallback = void Function(String path);
+
+class FilesGrid extends StatelessWidget {
```

Comment:
don't know if it's too wise to have a separate widget for this? mayube we could have something like a FileView with a delegate that builds the provided list of files. this would allow to not duplicate eventual code and to have also clean code (most stuff would be still separated with delegates)

Reply:
Ya we can create FileView which can take child and `WorkspaceView` as properties and incapsulate choose between GridView and TableView

Reply:
a builder approach could be another solution yea, tho the delegate could be better for some cases, we can discuss more in depth if you want